### ISSUE TITLE
Rstan won't compile

### ISSUE THREAD

**Issue by dgallay** [stan-dev/rstan#773]
#### Summary:
Rstan won’t compile. Rstan worked fine until I upgraded to R4.0.1 and RTools4.0. 

#### Description:
Here’s the error message I get when trying to compile: 

Error in file(con, "r") : cannot open the connection
In addition: Warning messages:
1: In system(cmd, intern = !verbose) :
  running command 'C:/PROGRA~1/R/R-4.0.1/bin/x64/R CMD SHLIB file2cd4445c123a.cpp 2> file2cd4445c123a.cpp.err.txt' had status 1
2: In file(con, "r") :
  cannot open file 'file2cd4445c123a.cpp.err.txt': No such file or directory
Error in sink(type = "output") : invalid connection
Error in sink(type = "output") : invalid connection

#### Reproducible Steps:
See attached .

If you have a reproducible example, please include it.

#### Current Output:
If applicable, any relevant output from RStan.

#### Expected Output:
See error message above.

#### RStan Version:
rstan_2.19.3   
StanHeaders_2.21.0-5
#### R Version:
R version 4.0.1 (2020-06-06)

#### Operating System:
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

[Simple_Regression1.zip](https://github.com/stan-dev/rstan/files/4755345/Simple_Regression1.zip)

**Comment by jgabry** [comment]
Sorry you've having compilation issues. I don't know Windows well enough to know what's going on but maybe this advice on the Stan forum will help: 

https://discourse.mc-stan.org/t/r-4-0-rstan-and-you/15271

If not, I would suggest starting a new topic on the forum because more people will see it there than here on github.

**Comment by jgabry** [comment]
@dgallay Were you ever able to get this fixed? If not, does this post on the Stan forums help? https://discourse.mc-stan.org/t/rstan-on-windows/16673

**Comment by dgallay** [comment]
No.  I think the problem is with my Windows configuration; not with Stan. Before I upgraded to R4.0, I had no issues with rstan.  Then, I started using One Drive to back up my files.  I think that’s the source of my problem.  Here’s why: I’m getting this  error message (see below).  I’m not computer literate, so I can’t fully decipher the error. But it seems that the program gets “confused” going from my hard-drive to One Drive.  I don’t have this problem with any other R program. So, I’m guessing I can’t pass my model to the C++ Toolchain software.  (At least, that’s my non-technical take on the situation. 

 

Closing One Drive before installing rstan has no impact.  And I’m deathly afraid about uninstalling my One Drive.  I’m not sure what files I might lose if I did that. 

 

Unless, I can figure out a fix (or a work-around), I’m going to have amend my Bayesian course in the fall.  However, I’m not particularly keen about reverting to jags.

 

Thanks for inquiring.

**Comment by jgabry** [comment]
Hmm, sorry to hear that. I don't know anything about One Drive unfortunately. If RStan doesn't end up working then Instead of reverting to JAGS, I would suggest using our other Stan interface for R, which is the brand new CmdStanR interface. It's easy to use and I'd be very curious to know if it works on Windows even with One Drive installed (it uses a different mechanism for interfacing with Stan than RStan does). Here are a few useful links if you're interested in trying that out:  

* [Special installation instructions for Windows](https://www.maxmantei.com/blog/2020-05-16-installing-cmdstanr-on-windows/cmdstanr-windows/)
* [CmdStanR tutorials](https://mc-stan.org/cmdstanr/articles/index.html)

I know it's been working for a lot of other Windows users who have made the transition from RStan to CmdStanR recently.

**Comment by breckbaldwin** [comment]
There are a few workarounds. 
1) Use CmdStanR. It has most of the functionality of RStan but works for most people. Jonah just made that suggestion while I was typing this.
2)  RStudio cloud allows you to run RStan in a cloud environment. They will start charging this August but prices are not bad. $5 student/month, instructor is $15/month. https://rstudio.cloud/plans/instructor. I have been playing with it, PM me if you want more info. fbb2116@columbia.edu

Breck

**Comment by dgallay** [comment]
Thanks!  I’ll give it a try.

**Comment by jgabry** [comment]
Great, definitely let us know if you run into any trouble and we’ll try to help out.
